pallets/liquidity-pools-gateway/queue/src/benchmarking.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use frame_benchmarking::{account, impl_benchmark_test_suite, v2::*};
+use frame_system::RawOrigin;
+use parity_scale_codec::EncodeLike;
+
+use super::*;
+
+#[benchmarks(
+	where
+		T: Config,
+		T::AccountId: EncodeLike<<T as frame_system::Config>::AccountId>,
+)]
+mod benchmarks {
+	use super::*;
+
+	#[benchmark]
+	fn process_message() -> Result<(), BenchmarkError> {
+		let caller: T::AccountId = account("acc_0", 0, 0);
+		let message = T::Message::default();
```

Comment:
I am a bit worried about the PoV with batch messages, but I think we can go with it now 👍🏻

Reply:
Maybe in the next steps we can have something like a `MaxEncodedLenSelf` trait or similar for cases like these.

Reply:
Yeah, something like:

```rust
#[pallet::weight(T::WeightInfo::process_message().saturating_add(Weights::from_parts(0, T::Message::max_encoded_len())))]
```